packages/url-state-provider/src/index.ts

```diff
@@ -58,19 +71,19 @@ function decode(string) {
 
   try {
     // try to decode as lz-string
-    return JSON.parse(LZString.decompressFromEncodedURIComponent(string))
+    const json: object = JSON.parse(LZString.decompressFromEncodedURIComponent(string))
+    return json
   } catch (e) {
     console.warn("URL State Provider: Could not decode string: ", string, e)
     return {}
   }
 }
 
 /**
- * find search param by key and convert it to json
- * @param {string} searchKey
+ * find search param by key (regex) and convert it to json
  * @returns json
  */
-function URLToState() {
+function URLToState(): { [key: string]: any } {
```

Comment:
I’m wondering if we could replace the any type with a more specific one, like this:

```typescript
type DecodedState = { [key: string]: string | number | boolean | any[] | null };

function URLToState(): DecodedState {
  // ...
```

This change would enhance type safety and provide better clarity regarding the structure of the returned object.
What do you think?